transports/daily/src/transport.ts

```diff
@@ -608,25 +619,82 @@ export class DailyTransport extends Transport {
     }
   }
 
+  private handleDeviceError(ev: DailyEventObjectCameraError) {
+    const generateDeviceError = (
+      error: DailyCameraErrorObject<DailyCameraErrorType>,
+    ) => {
+      const devices: DeviceArray = [];
+      switch (error.type) {
+        case "permissions": {
+          error.blockedMedia.forEach((d) => {
+            devices.push(d === "video" ? "cam" : "mic");
+          });
+          return new DeviceError(devices, error.type, error.msg, {
+            blockedBy: error.blockedBy,
+          });
+        }
+        case "not-found": {
+          error.missingMedia.forEach((d) => {
+            devices.push(d === "video" ? "cam" : "mic");
+          });
+          return new DeviceError(devices, error.type, error.msg);
+        }
+        case "constraints": {
+          error.failedMedia.forEach((d) => {
+            devices.push(d === "video" ? "cam" : "mic");
+          });
+          return new DeviceError(devices, error.type, error.msg, {
+            reason: error.reason,
+          });
+        }
+        case "cam-in-use": {
+          devices.push("cam");
+          return new DeviceError(devices, error.type, error.msg);
+        }
+        case "mic-in-use": {
+          devices.push("mic");
+          return new DeviceError(devices, error.type, error.msg);
+        }
+        case "cam-mic-in-use":
+        case "undefined-mediadevices":
+        case "unknown":
+        default: {
+          devices.push("cam");
+          devices.push("mic");
+          return new DeviceError(devices, error.type, error.msg);
+        }
+      }
+    };
+    this._callbacks.onDeviceError?.(generateDeviceError(ev.error));
+  }
+
   private async handleLocalAudioTrack(track: MediaStreamTrack) {
     if (this.state == "ready" || !this._bufferLocalAudioUntilBotReady) {
       return;
     }
     const status = this._mediaStreamRecorder.getStatus();
     switch (status) {
       case "ended":
-        await this._mediaStreamRecorder.begin(track);
-        await this.startRecording();
+        try {
+          await this._mediaStreamRecorder.begin(track);
+          await this.startRecording();
+        } catch (e) {
+          // void. nothing to do.
```

Comment:
Same question as before, but now I'm wondering, is the reason you think we don't need to do anything here because the recorder would internally trigger a device error?

Reply:
correct. the idea is that begin() triggers the callback and logs warnings/errors already.